plutus-ledger-api/src/PlutusLedgerApi/V3/Contexts.hs

```diff
@@ -0,0 +1,577 @@
+    a PlutusTx.== a'
+  StakePoolVoter a == StakePoolVoter a' =
+    a PlutusTx.== a'
+  _ == _ = Haskell.False
+
+data Vote
+  = VoteNo
+  | VoteYes
+  | Abstain
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+  deriving (Pretty) via (PrettyShow Vote)
+
+instance PlutusTx.Eq Vote where
+  {-# INLINEABLE (==) #-}
+  VoteNo == VoteNo   = Haskell.True
+  VoteYes == VoteYes = Haskell.True
+  Abstain == Abstain = Haskell.True
+  _ == _             = Haskell.False
+
+-- | Similar to TxOutRef, but for GovActions
+data GovernanceActionId = GovernanceActionId
+  { gaidTxId        :: V2.TxId
+  , gaidGovActionIx :: Haskell.Integer
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty GovernanceActionId where
+  pretty GovernanceActionId{..} =
+    vsep
+      [ "gaidTxId:" <+> pretty gaidTxId
+      , "gaidGovActionIx:" <+> pretty gaidGovActionIx
+      ]
+
+instance PlutusTx.Eq GovernanceActionId where
+  {-# INLINEABLE (==) #-}
+  GovernanceActionId a b == GovernanceActionId a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data VotingProcedure = VotingProcedure
+  { vpVote   :: Vote
+  , vpAnchor :: Haskell.Maybe Anchor
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty VotingProcedure where
+  pretty VotingProcedure{..} =
+    vsep
+      [ "vpVote:" <+> pretty vpVote
+      , "vpAnchor:" <+> pretty vpAnchor
+      ]
+
+instance PlutusTx.Eq VotingProcedure where
+  {-# INLINEABLE (==) #-}
+  VotingProcedure a b == VotingProcedure a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data Committee = Committee
+  { committeeMembers :: Map ColdCommitteeHash Haskell.Integer
+  -- ^ Committee members with epoch number when each of them expires
+  , committeeQuorum  :: PlutusTx.Rational
+  -- ^ Quorum of the committee that is necessary for a successful vote
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Committee where
+  pretty Committee{..} =
+    vsep
+      [ "committeeMembers:" <+> pretty committeeMembers
+      , "committeeQuorum:" <+> pretty committeeQuorum
+      ]
+
+instance PlutusTx.Eq Committee where
+  {-# INLINEABLE (==) #-}
+  Committee a b == Committee a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data Constitution = Constitution
+  { constitutionHash   :: PlutusTx.BuiltinByteString -- Balke2b_256 of some off-chain data
+  , constitutionScript :: Haskell.Maybe V2.ScriptHash
+  }
+  deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty Constitution where
+  pretty Constitution{..} =
+    vsep
+      [ "constitutionHash:" <+> pretty constitutionHash
+      , "constitutionScript:" <+> pretty constitutionScript
+      ]
+
+instance PlutusTx.Eq Constitution where
+  {-# INLINEABLE (==) #-}
+  Constitution a b == Constitution a' b' =
+    a PlutusTx.== a' PlutusTx.&& b PlutusTx.== b'
+
+data ProtocolVersion = ProtocolVersion
+  { pvMajor :: Haskell.Integer
+  , pvMinor :: Haskell.Integer
```

Comment:
We can perhaps reuse the existing `ProtocolVersion` type, but that one is faster due to `Int`. I don't know whether it matters or not.

Reply:
Let's keep them separate for now, they have different purposes.